mountpoint-s3-crt/src/s3/pool.rs

```diff
@@ -357,14 +359,16 @@ impl<Buffer: AsMut<[u8]>> CrtTicket<Buffer> {
     }
 }
 
+/// Return the buffer associated with a ticket.
 unsafe extern "C" fn ticket_claim<Buffer: AsMut<[u8]>>(mut ticket: *mut aws_s3_buffer_ticket) -> aws_byte_buf {
     // SAFETY: `ticket` was obtained through `Ticket::leak`.
     let ticket = unsafe { CrtTicket::<Buffer>::ref_mut_from_raw(&mut ticket) };
 
     // SAFETY: the CRT guarantees to only use the returned buffer while holding the ticket.
     let aws_byte_cursor { len, ptr } = unsafe { ticket.buffer.as_mut().as_aws_byte_cursor() };
 
-    // SAFETY: `ptr` is a valid buffer with capacity >= `size`.
+    // Build the `aws_byte_buf` required by the CRT from the pointer and length of the buffer.
+    // SAFETY: `ptr` is a valid buffer with capacity >= `len`.
     unsafe { aws_byte_buf_from_empty_array(ptr as *mut libc::c_void, len) }
```

Comment:
Can we add another comment here saying we're using `aws_byte_buf_from_empty_array` rather than `aws_byte_buf_from_array` because the memory we're returning is to be used in future operations rather than the current content being significant. It took me a few reads to realise